### ISSUE TITLE
SystemJs packages defaultExtention option normalization error

### ISSUE THREAD

**Issue by milkshakeuk** [systemjs/systemjs#527]
When using the new packages option in combination with defaultExtention I seem to get undesired normalization behaviour when using the text plugin.

```javascript
//File: config.js
System.config({
  "baseURL": "/",
  "defaultJSExtensions": true,
  "transpiler": "typescript",
  "paths": {
    "github:*": "jspm/github/*",
    "npm:*": "jspm/npm/*"
  }
});

System.config({
    "packages": {
        "js": {
        "main":"main.ts",
        "defaultExtension": "ts"
        }
    },
    "map": {
        "handlebars": "github:components/handlebars.js@3.0.3",
        "jquery": "github:components/jquery@2.1.4",
        "text": "github:systemjs/plugin-text@0.0.2",
        "typescript": "github:mhegazy/typescript@v1.5-beta2",
        "underscore": "npm:underscore@1.8.3"
    }
});
```

I try to use the text plugin like so:

```javascript
//File: NavigationView.ts
import template from '/js/app/templates/nav.html!text';
```

and when during the normalization process...

```javascript
//File: system.src.js
return Promise.all([
          loader.normalize(argumentName, parentName),
          loader.normalize(pluginName, parentName)
        ])
        .then(function(normalized) {
          argumentName = normalized[0];
          // argumentName = "http://localhost:8080/js/app/templates/nav.html"  CORRECT
          if (defaultExtension) 
            argumentName = argumentName.substr(0, argumentName.length - 3); 
          // argumentName = "http://localhost:8080/js/app/templates/nav.h" NOW INCORRECT
          return argumentName + '!' + normalized[1];
        });
```
and of course at this point when the text plugin attempts to pull in my template it will not find the file because it trimmed off the last 3 chars of the extension. how can I use the `defaultExtention` option
 without breaking the normalization of the text plugin arguments? is this a bug or a configuration issue on my part?

 [here](https://github.com/milkshakeuk/tsdemo/tree/jspm) is a branch with the code in question for testing.

hope you can help

**Comment by guybedford** [comment]
Thanks, this is a bug - the conditional should be `if (defaultExtension && argumentName.substr(0, argumentName.length - 3) == '.js')`.

**Comment by milkshakeuk** [comment]
It took me a while to figure out what was happening but I'm glad I could help.

**Comment by milkshakeuk** [comment]
I tested that one line change locally and it fixed the problem.

**Comment by guybedford** [comment]
Released in 0.18.2.
